### DISCUSSION TITLE
Retrieve does not work correctly with multi-root metadata

### DISCUSSION THREAD

**Discussion by pkozuchowski** [forcedotcom/salesforcedx-vscode#5764] [Ideas]
### Summary
When metadata is organized into packages, for example like this:
force-app/commons/classes/classX.cls
force-app/main/default/classes/classY.cls

When I right click and retrieve on classX, the file is not refreshed. Instead new file is created in 
force-app/main/default/classes/classX.cls

### Steps To Reproduce:
The project should be setup against sandbox, not scratch org.
1. Create folder structure as mentioned above
Some classes are in force-app/commons/classes
The rest is in /main/default/classes

2. Right Click and Retrieve on class in /commons/classes
3. Notice that new file is created in default package.

### Expected result
Existing class should be refreshed with retrieved metadata.

### Actual result
New file is created in default package.

**VS Code Version**:
1.40.1

**SFDX CLI Version**:
7.32.0

**OS and version**:
Windows 10.0.18362

**Comment by lcampos** [comment]
Hi @amorek thanks for opening this issue. We will be addressing this limitation soon and you will see it being added to our roadmap. 

The current `source:retrieve` behavior is limited by the configuration provided on the project's `sfdx-project.json` file where the command will always retrieve to the `path` set to being the default in `packageDirectories`. We will be enhancing the cli commands in order to address this limitation and be able to override that configuration.

cc @ntotten
